File: src/main/java/de/gesellix/docker/client/filesocket/UnixSocketFactorySupport.java
package de.gesellix.docker.client.filesocket;

import org.newsclub.net.unix.AFUNIXSocket;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class UnixSocketFactorySupport {

  private static final Logger log = LoggerFactory.getLogger(UnixSocketFactorySupport.class);

  public boolean isSupported() {
    return isSupported(System.getProperty("os.name"));
  }

  public boolean isSupported(String osName) {
    try {
      boolean isWindows = osName.toLowerCase().contains("windows");
      return !isWindows && AFUNIXSocket.isSupported();
    }
    catch (Throwable reason) {
      log.info("Unix socket not supported", reason);
      return false;
    }
  }
}

File: src/main/kotlin/de/gesellix/docker/client/filesocket/UnixSocketFactorySupport.kt
package de.gesellix.docker.client.filesocket

import mu.KotlinLogging
import org.newsclub.net.unix.AFUNIXSocket

private val log = KotlinLogging.logger {}

class UnixSocketFactorySupport {

    fun isSupported(): Boolean = isSupported(System.getProperty("os.name"))

    fun isSupported(osName: String): Boolean {
        return try {
            val isWindows = osName.toLowerCase().contains("windows")
            !isWindows && AFUNIXSocket.isSupported()
        } catch (reason: Throwable) {
            log.info("unix socket not supported", reason)
            false
        }
    }
}
